File: woodroid/src/main/java/me/gilo/woodroid/models/filters/ReportsDateFilter.java
package me.gilo.woodroid.models.filters;

public class ReportsDateFilter extends Filter{

    String period;
    String date_min;
    String date_max;

    public String getPeriod() {
        return period;
    }

    public void setPeriod(String period) {
        this.period = period;

        addFilter("period", period);
    }

    public String getDate_min() {
        return date_min;
    }

    public void setDate_min(String date_min) {
        this.date_min = date_min;

        addFilter("date_min", date_min);
    }

    public String getDate_max() {
        return date_max;
    }

    public void setDate_max(String date_max) {
        this.date_max = date_max;

        addFilter("date_max", date_max);
    }
}

File: core/src/main/java/me/gilo/woodroid/core/models/filters/ReportsDateFilter.kt
package me.gilo.woodroid.models.filters

class ReportsDateFilter : Filter() {

    internal lateinit var period: String
    internal lateinit var date_min: String
    internal lateinit var date_max: String

    fun getPeriod(): String {
        return period
    }

    fun setPeriod(period: String) {
        this.period = period

        addFilter("period", period)
    }

    fun getDate_min(): String {
        return date_min
    }

    fun setDate_min(date_min: String) {
        this.date_min = date_min

        addFilter("date_min", date_min)
    }

    fun getDate_max(): String {
        return date_max
    }

    fun setDate_max(date_max: String) {
        this.date_max = date_max

        addFilter("date_max", date_max)
    }
}
